### PR TITLE
allow onboarding follows to be local accounts

### PR DESCRIPTION
**Pull request by sdubinsky**

### CODE REVIEW

**Comment by sdubinsky** [comment]
tested on staging - works.

File: app/controllers/api/v1/onboarding_follow_recommendations_controller.rb [hunk 1]
```diff
@@ -20,12 +20,21 @@ def onboarding_follow_recommendations
...
           .reject { |a| a[:account].nil? }
       )
     end
   end
 
+  def find_account(account)
+    # TODO: this probably belongs in account_finder_concern.rb.
+    # I'm putting it here for now to avoid messing with mastodon code.
+    # if we need it elsewhere, we should move it. SD
+    username, domain = *username_and_domain(account['account'])
```
Comment (nvella) [inline app/controllers/api/v1/onboarding_follow_recommendations_controller.rb:33]: Love a nice splat *️⃣

Comment (sdubinsky) [reply app/controllers/api/v1/onboarding_follow_recommendations_controller.rb:33]: felipe's splat lol

File: app/controllers/api/v1/onboarding_follow_recommendations_controller.rb [hunk 2]
```diff
@@ -20,12 +20,21 @@ def onboarding_follow_recommendations
...
+    Account.find_remote(username, domain) ||
```
Comment (nvella) [inline app/controllers/api/v1/onboarding_follow_recommendations_controller.rb:34]: `Account.find_remote` may be passed `nil` for `domain` if a handle is provided without domain (eg `@foo`). Is this OK, or should we be doing a conditional on `domain.nil?`

**Review by nvella — APPROVED** [review]

File: app/controllers/api/v1/onboarding_follow_recommendations_controller.rb (hunk 2 above)
Comment (sdubinsky) [reply app/controllers/api/v1/onboarding_follow_recommendations_controller.rb:34]: `Account.find_local` works by calling `find_remote` and passing `nil` as the domain, so it's fine.  Good catch though.

**Pull request merged**